release_notes/utils.py

```diff
@@ -0,0 +1,52 @@
+import json
+from typing import Optional
+
+import git
+from github3 import GitHub, pulls, exceptions
+from github3.structs import GitHubIterator
+
+
+# pylint: disable=protected-access
+# noinspection PyProtectedMember
+def list_associated_pulls(gh: GitHub, owner: str, repo: str, sha: str) -> Optional[tuple[pulls.ShortPullRequest]]:
+    """ Returns a tuple with pull requests related to the specified commit.
+
+    :param gh: Instance of the GitHub v3 API
+    :param owner: Owner of the repository (on GitHub)
+    :param repo: Name of the repository (on GitHub)
+    :param sha: SHA of the commit
+    :return: a tuple with pull requests related to the specific commit
+    """
+    try:
+        url = gh._build_url("repos", owner, repo, "commits", sha, "pulls")
+        return tuple(gh._iter(-1, url, pulls.ShortPullRequest))
+    except exceptions.UnprocessableEntity as e:
+        if e.code != 422:  # pull request not found
```

Comment:
when will this happen?

Reply:
when the requested commit was not found in the repository. I've updated the comment.

Reply:
+1 for the updated comment. however: can this actually happen? You retrieve pull-requests immediately above 🤔

Reply:
Good question. I don't think this can actually happen in production. It's mostly a leftover from debugging at this point, where I got that error often when using dummy data 😅